From be74bbf7aaa00c828fd9e01b8d3e25a42ea0d129 Mon Sep 17 00:00:00 2001
From: Bruno Martins <bgcngm@gmail.com>
Date: Mon, 4 Dec 2017 23:49:07 +0000
Subject: [PATCH 1/4] PrivacyGuard: Bring up and inject into Settings

 * Leave Protected Apps disabled until its bring up is finished

Change-Id: I433f9f619c472ffc4600836be4011d7f13f932e9
---
 AndroidManifest.xml                                | 20 +++++------------
 res/layout/privacy_guard_manager_list_row.xml      | 26 +++++++++++++++-------
 res/xml/privacy_settings.xml                       |  2 ++
 .../lineageparts/privacyguard/AppInfoLoader.java   |  6 +----
 .../privacyguard/PrivacyGuardAppListAdapter.java   |  7 ------
 .../privacyguard/PrivacyGuardManager.java          |  2 --
 6 files changed, 27 insertions(+), 36 deletions(-)

diff --git a/AndroidManifest.xml b/AndroidManifest.xml
index f27ae25..7275ddc 100644
--- a/AndroidManifest.xml
+++ b/AndroidManifest.xml
@@ -103,31 +103,23 @@
                 android:resource="@string/live_display_summary" />
         </activity-alias>
 
-        <!-- Privacy settings (dashboard) -->
-        <!--
+        <!-- Privacy settings (Security & Location category) -->
         <activity-alias
             android:name=".PrivacySettings"
             android:label="@string/privacy_settings_title"
             android:targetActivity="PartsActivity">
-            <intent-filter android:priority="3">
-                <action android:name="com.android.settings.action.EXTRA_SETTINGS" />
+            <intent-filter android:priority="1">
+                <action android:name="org.lineageos.lineageparts.action.SETTINGS" />
                 <action android:name="org.lineageos.lineageparts.PRIVACY_SETTINGS" />
                 <category android:name="android.intent.category.DEFAULT" />
             </intent-filter>
             <meta-data
                 android:name="com.android.settings.category"
-                android:value="com.android.settings.category.personal" />
-            <meta-data
-                android:name="com.android.settings.icon"
-                android:resource="@drawable/ic_settings_privacy" />
-            <meta-data
-                android:name="org.lineageos.settings.summary.receiver"
-                android:value="org.lineageos.lineageparts.PartsUpdater" />
+                android:value="com.android.settings.category.ia.security" />
             <meta-data
-                android:name="org.lineageos.settings.summary.key"
-                android:value="privacy_settings" />
+                android:name="com.android.settings.summary"
+                android:value="@string/summary_empty" />
         </activity-alias>
-        -->
 
         <!-- Button settings (System category) -->
         <activity-alias
diff --git a/res/layout/privacy_guard_manager_list_row.xml b/res/layout/privacy_guard_manager_list_row.xml
index eb6a327..bfb239a 100644
--- a/res/layout/privacy_guard_manager_list_row.xml
+++ b/res/layout/privacy_guard_manager_list_row.xml
@@ -15,6 +15,7 @@
 -->
 
 <GridLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
     android:layout_width="match_parent"
     android:layout_height="wrap_content"
     android:minHeight="?android:attr/listPreferredItemHeight"
@@ -22,14 +23,23 @@
     android:paddingEnd="?android:attr/listPreferredItemPaddingEnd"
     android:paddingTop="8dip"
     android:paddingBottom="8dip" >
-    <ImageView
-        android:id="@+id/app_icon"
-        android:layout_width="36dip"
-        android:layout_height="36dip"
-        android:layout_marginEnd="8dip"
-        android:layout_gravity="center_vertical"
-        android:scaleType="centerInside"
-        android:contentDescription="@null" />
+    <LinearLayout
+        android:id="@+id/icon_frame"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:minWidth="56dp"
+        android:gravity="start|center_vertical"
+        android:orientation="horizontal"
+        android:paddingRight="8dp"
+        android:paddingTop="4dp"
+        android:paddingBottom="4dp">
+        <android.support.v7.internal.widget.PreferenceImageView
+            android:id="@+id/app_icon"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            app:maxWidth="48dp"
+            app:maxHeight="48dp" />
+    </LinearLayout>
     <TextView
         android:id="@+id/app_title"
         android:layout_width="0dip"
diff --git a/res/xml/privacy_settings.xml b/res/xml/privacy_settings.xml
index c3fb516..5f4034f 100644
--- a/res/xml/privacy_settings.xml
+++ b/res/xml/privacy_settings.xml
@@ -27,6 +27,7 @@
         android:fragment="org.lineageos.lineageparts.privacyguard.PrivacyGuardManager" />
 
     <!-- Protected apps -->
+    <!-- 
     <Preference
         android:title="@string/protected_apps_manager_title"
         android:summary="@string/protected_apps_manager_summary">
@@ -35,6 +36,7 @@
             android:targetPackage="com.android.settings"
             android:targetClass="com.android.settings.applications.ProtectedAppsActivity" />
     </Preference>
+    -->
 
     <!-- Anonymous statistics - (LineageStats) -->
     <PreferenceScreen
diff --git a/src/org/lineageos/lineageparts/privacyguard/AppInfoLoader.java b/src/org/lineageos/lineageparts/privacyguard/AppInfoLoader.java
index bd2bc49..b7658c0 100644
--- a/src/org/lineageos/lineageparts/privacyguard/AppInfoLoader.java
+++ b/src/org/lineageos/lineageparts/privacyguard/AppInfoLoader.java
@@ -22,9 +22,7 @@ import android.content.pm.ApplicationInfo;
 import android.content.pm.PackageManager;
 import android.content.pm.PackageInfo;
 
-/*
 import org.lineageos.lineageparts.privacyguard.PrivacyGuardManager.AppInfo;
-*/
 
 import java.util.ArrayList;
 import java.util.Collections;
@@ -34,7 +32,7 @@ import java.util.List;
 /**
  * An asynchronous loader implementation that loads AppInfo structures.
  */
-/* package *//* class AppInfoLoader extends AsyncTaskLoader<List<AppInfo>> {
+/* package */ class AppInfoLoader extends AsyncTaskLoader<List<AppInfo>> {
     private PackageManager mPm;
     private boolean mShowSystemApps;
     private AppOpsManager mAppOps;
@@ -84,7 +82,6 @@ import java.util.List;
     *
     * @return the complete List off installed applications (@code PrivacyGuardAppInfo)
     */
-    /*
     private List<AppInfo> loadInstalledApps() {
         List<AppInfo> apps = new ArrayList<AppInfo>();
         List<PackageInfo> packages = mPm.getInstalledPackages(
@@ -125,4 +122,3 @@ import java.util.List;
     }
 
 }
-*/
diff --git a/src/org/lineageos/lineageparts/privacyguard/PrivacyGuardAppListAdapter.java b/src/org/lineageos/lineageparts/privacyguard/PrivacyGuardAppListAdapter.java
index 50db95f..d08747e 100644
--- a/src/org/lineageos/lineageparts/privacyguard/PrivacyGuardAppListAdapter.java
+++ b/src/org/lineageos/lineageparts/privacyguard/PrivacyGuardAppListAdapter.java
@@ -30,15 +30,12 @@ import android.widget.SectionIndexer;
 import android.widget.TextView;
 
 import org.lineageos.lineageparts.R;
-/*
 import org.lineageos.lineageparts.privacyguard.PrivacyGuardManager.AppInfo;
-*/
 
 import java.util.Arrays;
 import java.util.List;
 import java.util.concurrent.ConcurrentHashMap;
 
-/*
 public class PrivacyGuardAppListAdapter extends BaseAdapter implements SectionIndexer {
 
     private LayoutInflater mInflater;
@@ -145,7 +142,6 @@ public class PrivacyGuardAppListAdapter extends BaseAdapter implements SectionIn
          * To get from that number to the expected value of 1 we need to negate
          * and subtract 2.
          */
-        /*
         return index >= 0 ? index : -index - 2;
     }
 
@@ -157,7 +153,6 @@ public class PrivacyGuardAppListAdapter extends BaseAdapter implements SectionIn
     /**
      * An asynchronous task to load the icons of the installed applications.
      */
-    /*
     private class LoadIconsTask extends AsyncTask<PrivacyGuardManager.AppInfo, Void, Void> {
         @Override
         protected Void doInBackground(PrivacyGuardManager.AppInfo... apps) {
@@ -183,11 +178,9 @@ public class PrivacyGuardAppListAdapter extends BaseAdapter implements SectionIn
     /**
      * App view holder used to reuse the views inside the list.
      */
-    /*
     public static class PrivacyGuardAppViewHolder {
         TextView title;
         ImageView icon;
         ImageView privacyGuardIcon;
     }
 }
-*/
diff --git a/src/org/lineageos/lineageparts/privacyguard/PrivacyGuardManager.java b/src/org/lineageos/lineageparts/privacyguard/PrivacyGuardManager.java
index 6e7e908..c23d42b 100644
--- a/src/org/lineageos/lineageparts/privacyguard/PrivacyGuardManager.java
+++ b/src/org/lineageos/lineageparts/privacyguard/PrivacyGuardManager.java
@@ -54,7 +54,6 @@ import org.lineageos.lineageparts.SettingsPreferenceFragment;
 import java.util.ArrayList;
 import java.util.List;
 
-/*
 public class PrivacyGuardManager extends SettingsPreferenceFragment
         implements OnItemClickListener, OnItemLongClickListener,
                    LoaderManager.LoaderCallbacks<List<PrivacyGuardManager.AppInfo>> {
@@ -395,4 +394,3 @@ public class PrivacyGuardManager extends SettingsPreferenceFragment
         }
     }
 }
-*/
-- 
2.7.4

